src/Marten/Linq/Parsing/SimpleEqualsParser.cs

```diff
@@ -4,99 +4,106 @@
-		{
-			var field = GetField(mapping, expression);
-			var locator = field.SqlLocator;
-
-			var value = expression.Arguments.OfType<ConstantExpression>().FirstOrDefault();
-			if (value == null) throw new BadLinqExpressionException("Could not extract value from {0}.".ToFormat(expression), null);
-
-			object valueToQuery = value.Value;
-
-			if (valueToQuery == null)
-			{
-				return new WhereFragment($"({locator}) {_isOperator} null");
-			}
-
-			if (valueToQuery.GetType() != expression.Method.DeclaringType)
-			{
-				try
-				{
-					valueToQuery = Convert.ChangeType(value.Value, expression.Method.DeclaringType);
-				}
-				catch (Exception e)
-				{
-					throw new BadLinqExpressionException(
-						$"Could not convert {value.Value.GetType().FullName} to {expression.Method.DeclaringType}", e);
-				}
-			}
-
-			if (_supportContainment && ((mapping.PropertySearching == PropertySearching.ContainmentOperator ||
-			                             field.ShouldUseContainmentOperator()) &&
-			                            !(field is DuplicatedField)))
-			{
-				var dict = new Dictionary<string, object>();
-				ContainmentWhereFragment.CreateDictionaryForSearch(dict, expression, valueToQuery, serializer);
-				return new ContainmentWhereFragment(serializer, dict);
-			}
-
-			return new WhereFragment($"{locator} {_equalsOperator} ?", valueToQuery);
-		}
-
-		private static IField GetField(IQueryableDocument mapping, MethodCallExpression expression)
-		{
-			IField GetField(Expression e)
-			{
-				var visitor = new FindMembers();
-				visitor.Visit(e);
-
-				var field = mapping.FieldFor(visitor.Members);
-				return field;
-			}
+{
+    /// <summary>
+    /// Implement Equals for <see cref="int"/>, <see cref="long"/>, <see cref="decimal"/>, <see cref="Guid"/>, <see cref="bool"/>, <see cref="DateTime"/>, <see cref="DateTimeOffset"/>.
+    /// </summary>
+    /// <remarks>Equals(object) calls into <see cref="Convert.ChangeType(object, Type)"/>. Equals(null) is converted to "is null" query.</remarks>
+    public class SimpleEqualsParser : IMethodCallParser
+    {
+        private static readonly List<Type> SupportedTypes = new List<Type> {
+            typeof(int), typeof(long), typeof(decimal), typeof(Guid), typeof(bool)
+        };
+
+        private readonly string _equalsOperator;
+        private readonly string _isOperator;
+        private readonly bool _supportContainment;
 
-			if (!expression.Method.IsStatic && expression.Object != null && expression.Object.NodeType != ExpressionType.Constant)
-			{
-				// x.member.Equals(...)
-				return GetField(expression.Object);
-			}
-			if (expression.Arguments[0].NodeType == ExpressionType.Constant)
-			{
-				// type.Equals("value", x.member) [decimal]
-				return GetField(expression.Arguments[1]);
-			}
-			// type.Equals(x.member, "value") [decimal]
-			return GetField(expression.Arguments[0]);
-		}
-	}
+        static SimpleEqualsParser()

```

Comment:
I don't know why this file diff is looking so odd. I just added this constructor

Reply:
yeah, the diff looks weird, difficult to review changes